### PR TITLE
Feature/rn example

### PR DESCRIPTION
**Pull request by mayteio**
Opening early to centralise discussion & fixes.

### CODE REVIEW

File: example-rn/__tests__/Example.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  // render component and provider with a mocked context value
+  const {findByText} = render(
+    <PusherProvider clientKey="client-key" cluster="ap4" value={{client}}>
+      <Example />
+    </PusherProvider>,
+  );
+
+  // emit an event on the mocked channel
+  act(() => mockChannel.emit('title', {data: 'Hello world'}));
```
Comment (hsavit1) [inline example-rn/__tests__/Example.js:52]: instead of an `act` , this should probably be a `waitFor`

Comment (hsavit1) [inline example-rn/__tests__/Example.js:52]:
```suggestion
  await waitFor(() => mockChannel.emit('title', {data: 'Hello world'}));
```

File: example-rn/__tests__/Example.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  PusherProvider,
+  useChannel,
+  useEvent,
+  useState,
+} from '@harelpls/use-pusher/react-native';
+
+import {View, Text} from 'react-native';
+
+// Example.test.tsx
+import {render, act} from 'react-native-testing-library';
```
Comment (hsavit1) [inline example-rn/__tests__/Example.js:13]:
```suggestion
import {render, act, waitFor} from 'react-native-testing-library';
```

File: example-rn/__tests__/Example.js (hunk 1 above)
Comment (mayteio) [reply example-rn/__tests__/Example.js:52]: Hmmm... not sure what this solves? The tests pass. Additionally, `PusherChannelMock.emit` doesn't return a promise, and we're doing 'non-react things' that react needs to know about, 

> To prepare a component for assertions, wrap the code rendering it and performing updates inside an act() call. This makes your test run closer to how React works in the browser.

Comment (hsavit1) [reply example-rn/__tests__/Example.js:52]: oh, then please ignore! I admittedly had not run the tests. Looking at this again, I had no idea what I was thinking!

**Pull request closed without merging**